### DISCUSSION TITLE
Unexpected behavior of "chunks" argument in open_mfdataset()

### DISCUSSION THREAD

**Discussion by arthur-e** [pydata/xarray#9119] [General] [answered]
I'm confused as to why using `xr.open_mfdataset(..., chunks = {'time': N})` is not producing chunks with `N` elements along the `time` dimension.

An example dataset can be obtained using `earthaccess`, with [NASA EarthData Search credentials.](https://search.earthdata.nasa.gov/search)

```python
import earthaccess
import xarray as xr
auth = earthaccess.login()

results = earthaccess.search_data(
    short_name = 'M2SDNXSLV',
    temporal = ("2024-01-01", "2024-05-31"))

# Could take about 1 minute on a broadband connection
earthaccess.download(results, 'data_raw/MERRA2')
```

This example easily fits into memory, but is an example for a tutorial I'm working on. A more salient reason for wanting a specific kind of chunking would be, e.g., a desire to calculate long-term trends, so all the elements along the `time` dimension should be in the same chunk.

Specifically, when `xarray` uses automatic chunking...

```python
ds = xr.open_mfdataset('./data_raw/MERRA2/*.nc4', chunks = 'auto')
ds['T2MMEAN'].chunksizes
```
```
Frozen({'time': (1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1), 'lat': (361,), 'lon': (576,)})
```

It creates a separate chunk for each file (each time step), which is not ideal. If I instead try:

```python
# This doesn't give the desired result
ds = xr.open_mfdataset('./data_raw/MERRA2/*.nc4', chunks = {'time': 122})

# Nor does this
ds = xr.open_mfdataset('./data_raw/MERRA2/*.nc4', chunks = {'lat': 91, 'lon': 144, 'time': 122})
```

I still get chunks that do not have 122 elements along the `time` dimension. The only way I can make this work is to rechunk the data after loading, [which is specifically called out as a bad practice.](https://docs.xarray.dev/en/stable/user-guide/dask.html#using-dask-with-xarray)

```python
# This finally does it
ds.chunk({'time': 122})
```

**Comment by dcherian** [comment, marked as answer]
The `chunks` argument is applied on a per-file basis so this is expected.


We'd happily merge a PR making this point clear in the documentation: https://github.com/pydata/xarray/blob/9237f90d35a64561aba5117c66cfbd43839528f8/xarray/backends/api.py#L861-L866
